File: generate-icons.js
import { createCanvas } from 'canvas';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Icon sizes needed for Tauri
const sizes = [32, 128, 256, 512];

function generateIcon(size) {
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');

  // Fill background with dark color (#1a1a1a)
  ctx.fillStyle = '#1a1a1a';
  ctx.fillRect(0, 0, size, size);

  // Set text properties
  ctx.fillStyle = '#00ff00'; // Green color
  ctx.font = `bold ${size * 0.75}px Arial, sans-serif`;
  ctx.textAlign = 'center';
  ctx.textBaseline = 'middle';

  // Draw the navigation symbol ⋋
  ctx.fillText('⋋', size / 2, size / 2);

  return canvas;
}

// Generate icons for all sizes
sizes.forEach(size => {
  const canvas = generateIcon(size);
  const buffer = canvas.toBuffer('image/png');
  const filename = size === 32 ? '32x32.png' : size === 128 ? '128x128.png' : `${size}x${size}.png`;
  const filepath = path.join(__dirname, 'src-tauri', 'icons', filename);

  fs.writeFileSync(filepath, buffer);
  console.log(`Generated ${filename}`);
});

// Also generate the main icon.png (using 128x128 as base)
const mainCanvas = generateIcon(128);
const mainBuffer = mainCanvas.toBuffer('image/png');
fs.writeFileSync(path.join(__dirname, 'src-tauri', 'icons', 'icon.png'), mainBuffer);
console.log('Generated icon.png');

// Generate 128x128@2x.png for retina displays
const retinaCanvas = generateIcon(256); // 128 * 2
const retinaBuffer = retinaCanvas.toBuffer('image/png');
fs.writeFileSync(path.join(__dirname, 'src-tauri', 'icons', '128x128@2x.png'), retinaBuffer);
console.log('Generated 128x128@2x.png');

console.log('All icons generated successfully!');

File: generate-icons.py
#!/usr/bin/env python3
from PIL import Image, ImageDraw, ImageFont
import os

# Icon sizes needed for Tauri
sizes = [32, 128, 256, 512]

def generate_icon(size):
    # Create image with dark background
    img = Image.new('RGBA', (size, size), color='#1a1a1a')
    draw = ImageDraw.Draw(img)
    
    # Calculate font size (70% of image size)
    font_size = int(size * 0.7)
    
    try:
        # Try to use a system font that supports the navigation symbol
        font = ImageFont.truetype('/System/Library/Fonts/Supplemental/Arial Unicode.ttf', font_size)
    except:
        try:
            font = ImageFont.truetype('/System/Library/Fonts/Helvetica.ttc', font_size)
        except:
            # Fallback to default font
            font = ImageFont.load_default()
    
    # Navigation symbol ⋋
    text = '⋋'
    
    # Get text bounding box for centering
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]
    
    # Calculate position to center the text
    x = (size - text_width) // 2 - bbox[0]
    y = (size - text_height) // 2 - bbox[1]
    
    # Draw the navigation symbol in bright green
    draw.text((x, y), text, fill='#00ff00', font=font)
    
    return img

# Ensure icons directory exists
icons_dir = 'src-tauri/icons'
os.makedirs(icons_dir, exist_ok=True)

# Generate icons for all sizes
for size in sizes:
    img = generate_icon(size)
    filename = f'{size}x{size}.png'
    filepath = os.path.join(icons_dir, filename)
    img.save(filepath, 'PNG')
    print(f'✓ Generated {filename}')

# Generate specific named files
specific_files = [
    (32, '32x32.png'),
    (128, '128x128.png'),
    (128, 'icon.png'),
    (256, '128x128@2x.png')
]

for size, name in specific_files:
    img = generate_icon(size)
    filepath = os.path.join(icons_dir, name)
    img.save(filepath, 'PNG')
    print(f'✓ Generated {name}')

print('\n✅ All icons generated successfully with navigation symbol ⋋!')
